src/components/v2/devtronStackManager/DevtronStackManager.component.tsx

```diff
@@ -429,91 +429,214 @@ export const InstallationWrapper = ({
+    const [preRequisiteList, setPreRequisiteList] = useState<
+        { version: string; prerequisiteMessage: string; tagLink: string }[]
+    >([])
+    useEffect(() => {
+        if (releaseNotes) {
+            const _preRequisiteList = []
+            for (let index = 0; index < releaseNotes.length; index++) {
+                const element = releaseNotes[index]
+                if (serverInfo && serverInfo.currentVersion === element.releaseName) {
+                    break
+                }
+                if (element.prerequisite && element.prerequisiteMessage) {
+                    _preRequisiteList.push({
+                        version: element.releaseName,
+                        prerequisiteMessage: element.prerequisiteMessage,
+                        tagLink: element.tagLink,
+                    })
+                }
+            }
+            setPreRequisiteList(_preRequisiteList.reverse())
+        }
+    }, [releaseNotes])
 
     const handleActionButtonClick = () => {
         if (isActionTriggered) {
             return
         } else {
-            updateActionTrigger(true)
-            handleAction(moduleName, isUpgradeView, upgradeVersion, updateActionTrigger, history, location)
+            if (preRequisiteChecked || preRequisiteList.length === 0) {
+                setShowPreRequisiteConfirmationModal(false)
+                updateActionTrigger(true)
+                handleAction(moduleName, isUpgradeView, upgradeVersion, updateActionTrigger, history, location)
+            } else {
+                setShowPreRequisiteConfirmationModal(true)
+            }
         }
     }
 
-    return (
-        <div className="module-details__install-wrapper">
-            {serverInfo?.installationType && serverInfo.installationType !== InstallationType.OSS_HELM ? (
-                <>
-                    {serverInfo.installationType === InstallationType.OSS_KUBECTL && (
-                        <NotSupportedNote isUpgradeView={isUpgradeView} />
-                    )}
-                    {serverInfo.installationType === InstallationType.ENTERPRISE && <ManagedByNote />}
-                </>
-            ) : (
-                <>
-                    {installationStatus !== ModuleStatus.INSTALLING &&
-                        installationStatus !== ModuleStatus.UPGRADING &&
-                        installationStatus !== ModuleStatus.INSTALLED &&
-                        (installationStatus !== ModuleStatus.HEALTHY ||
-                            (installationStatus === ModuleStatus.HEALTHY && latestVersionAvailable)) && (
-                            <button
-                                className="module-details__install-button cta flex mb-16"
-                                onClick={handleActionButtonClick}
+    const handlePrerequisiteCheckedChange = (): void => {
+        setPreRequisiteChecked(!preRequisiteChecked)
+    }
+
+    const renderConfirmationModal = (): JSX.Element | null => {
```

Comment:
Can we name it as  `renderPrerequisiteConfirmationModal`

Reply:
done